apps/VendingMachine/test/VendingMachine.test.ts

```diff
@@ -0,0 +1,230 @@
+      attoAlphAmount: DUST_AMOUNT
+    })
+
+    console.log('New Base URI: ', hexToString((await vendingMachineStates.methods.getBaseUri()).returns))
+  })
+
+  it('should Toggle Mint state ', async () => {
+    const vendingMachineStates = VendingMachine.at(vendingMachineContractAddress)
+
+    await ToggleMintState.execute(signer, {
+      initialFields: {
+        vendingMachine: vendingMachineContractAddress
+      },
+      attoAlphAmount: DUST_AMOUNT
+    })
+
+    console.log('Is Mint Paused: ', (await vendingMachineStates.methods.isMintPaused()).returns)
+  })
+
+  it('should Mint an NFT', async () => {
+    const vendingMachineStates = VendingMachine.at(vendingMachineContractAddress)
+    let mintPrice = (await vendingMachineStates.methods.getMintPrice()).returns
+    console.log(`Mint Price: ${mintPrice / BigInt(NO_DECIMALS)} ALPH`)
+    let mintAmount = 10n
+    console.log('Total Mint: ', mintAmount)
+    let alphNeededForMint = mintPrice * mintAmount
+    let alphNeededForSubcontract = mintAmount * ONE_ALPH + mintAmount * DUST_AMOUNT
+    let totalAlphNeeded = alphNeededForMint + alphNeededForSubcontract
+    console.log('Signer 1 owner: ', signer1.address)
+    console.log('Signer 1 ALPH balance: ', Number(await getAlphBalance(signer1.address, signer)) / NO_DECIMALS)
+    console.log(
+      'Foods Collection ALPH balance before Mint',
+      Number(await getAlphBalance(vendingMachineStates.address, signer)) / NO_DECIMALS
+    )
+    let newNftCreated = await MintNft.execute(signer1, {
+      initialFields: {
+        vendingMachine: vendingMachineContractAddress,
+        foodTypeId: 2n,
+        mintAmount: mintAmount
+      },
+      attoAlphAmount: totalAlphNeeded
+    })
+
+    console.log('user sent alph amount ', totalAlphNeeded / 1000000000000000000n)
+
+    console.log('Signer1 ALPH balance after mint', Number(await getAlphBalance(signer1.address, signer)) / NO_DECIMALS)
+    console.log(
+      'Vending machine Collection ALPH balance after Mint',
+      Number(await getAlphBalance(vendingMachineStates.address, signer)) / NO_DECIMALS
+    )
+
+    let returnedTransaction = newNftCreated.txId
+
+    let contractAddresses = await axios.get(
+      `http://127.0.0.1:22973/transactions/details/${returnedTransaction}?fromGroup=0&toGroup=0`
+    )
+    contractAddresses = contractAddresses.data.generatedOutputs
+    for (let i = 0; i < Number(mintAmount); i++) {
+      let newNftContractAddress = contractAddresses[i].address
+      const newNftContractState = Foods.at(newNftContractAddress)
+      console.log('NFT Index: ', (await newNftContractState.methods.getNFTIndex()).returns)
+    }
+  })
+
+  it('should  check user NFT balance ', async () => {
+    console.log('Signer 1 Address ', signer1.address)
+    console.log('Signer 1 ALPH balance: ', Number(await getAlphBalance(signer1.address, signer)) / NO_DECIMALS)
+    console.log('User NFT balance', await getTokenBalance(signer1.address, foodsContractAddress, signer))
```

Comment:
does this work properly? i think you are passing the token `address` here, but line 224 you are comparing the address to the `id` 
```ts
const tokenBalance = allBalances.tokenBalances?.find((t) => t.id === token)?.amount
```

Reply:
passed Id and worked